src/client/connector.rs

```diff
@@ -293,71 +253,32 @@ impl SystemService for ClientConnector {}
 
 impl Default for ClientConnector {
     fn default() -> ClientConnector {
-        #[cfg(all(feature = "alpn"))]
-        {
-            let builder = SslConnector::builder(SslMethod::tls()).unwrap();
-            ClientConnector::with_connector(builder.build())
-        }
-        #[cfg(all(feature = "tls", not(feature = "alpn")))]
-        {
-            let (tx, rx) = mpsc::unbounded();
-            let builder = NativeTlsConnector::builder();
-            ClientConnector {
-                stats: ClientConnectorStats::default(),
-                subscriber: None,
-                acq_tx: tx,
-                acq_rx: Some(rx),
-                resolver: None,
-                connector: builder.build().unwrap().into(),
-                conn_lifetime: Duration::from_secs(75),
-                conn_keep_alive: Duration::from_secs(15),
-                limit: 100,
-                limit_per_host: 0,
-                acquired: 0,
-                acquired_per_host: HashMap::new(),
-                available: HashMap::new(),
-                to_close: Vec::new(),
-                waiters: Some(HashMap::new()),
-                wait_timeout: None,
-                paused: Paused::No,
-            }
-        }
-        #[cfg(
-            all(
-                feature = "rust-tls",
-                not(any(feature = "alpn", feature = "tls"))
-            )
-        )]
-        {
-            let mut config = ClientConfig::new();
-            config
-                .root_store
-                .add_server_trust_anchors(&webpki_roots::TLS_SERVER_ROOTS);
-            ClientConnector::with_connector(config)
-        }
+        let connector = {
+            #[cfg(all(feature = "alpn"))]
+            { SslConnector::builder(SslMethod::tls()).unwrap().build() }
 
-        #[cfg(not(any(feature = "alpn", feature = "tls", feature = "rust-tls")))]
-        {
-            let (tx, rx) = mpsc::unbounded();
-            ClientConnector {
-                stats: ClientConnectorStats::default(),
-                subscriber: None,
-                acq_tx: tx,
-                acq_rx: Some(rx),
-                resolver: None,
-                conn_lifetime: Duration::from_secs(75),
-                conn_keep_alive: Duration::from_secs(15),
-                limit: 100,
-                limit_per_host: 0,
-                acquired: 0,
-                acquired_per_host: HashMap::new(),
-                available: HashMap::new(),
-                to_close: Vec::new(),
-                waiters: Some(HashMap::new()),
-                wait_timeout: None,
-                paused: Paused::No,
+            #[cfg(all(feature = "tls", not(feature = "alpn")))]
+            { NativeTlsConnector::builder().build().unwrap().into() }
```

Comment:
It is a bit not nice looking, I would suggest to not inline {} with expression like that

Reply:
that's fair, I am generally not too happy with the syntax around `#[cfg(..)]`. I wanted to omit the braces but apparently I can't do that for exprs

Reply:
yeah, sadly this is limitation of `cfg` I usually prefer to make it clear by creating visible scope
```
#[cfg]
{
    <expr>
}
```